### PR TITLE
 Fixes #405: added support to take the doc path of the sub projects from yaml

### LINKED ISSUE #405: Source for documentation of subprojects is taken as `docs`
### This is a
- [x] Bug report
- [ ] Feature request

### Description
I generated the documentation for https://github.com/BrimeNotes/documentation/ and added other projects as it sub projects. Generating docs from the homepage form and previewing it show the sub projects and their links correctly http://yaydoc.org/preview/ujjwalb1996@gmail.com/5cd9fe88-77d6-4002-b214-3207f616c59d_preview/
But when I register it to the CI and push docs on a commit, I don't see them there https://brimenotes.github.io/documentation/
* Here are the docs for CI http://yaydoc.org/BrimeNotes/documentation

```
/usr/src/app/temp/ujjwalb1996@gmail.com/957e202c-0ad0-44e5-b6c2-848e6a414a6b/yaydoctemp/index.rst:9: WARNING: toctree contains reference to nonexisting document u'android/docs/index'
/usr/src/app/temp/ujjwalb1996@gmail.com/957e202c-0ad0-44e5-b6c2-848e6a414a6b/yaydoctemp/index.rst:17: WARNING: toctree contains reference to nonexisting document u'core/docs/index'
/usr/src/app/temp/ujjwalb1996@gmail.com/957e202c-0ad0-44e5-b6c2-848e6a414a6b/yaydoctemp/index.rst:25: WARNING: toctree contains reference to nonexisting document u'webapp/docs/index'
```
Looking at the logs it appears that CI is taking source as docs/
i've defined . as the source of my subprojects in the `.yaydoc.yml` file of documentation repository.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

The source for sub projects in ci will be taken from the yaml file
## Related Issue

close #405 
## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
Can we review this PR?

**Comment by pri22296** [comment]
@sch00lb0y Please label this WIP if it's WIP.

**Comment by sch00lb0y** [comment]
@imujjwal96 @pri22296 please check http://demo08.herokuapp.com/

**Comment by imujjwal96** [comment]
I don't think it is working. Check the docs here: http://brime.ujjwalbhardwaj.me/
Repository: https://github.com/BrimeNotes/documentation/

File: backend/generator.js [hunk 1]
```diff
@@ -25,10 +25,15 @@ exports.executeScript = function (socket, formData, callback) {
   var docPath = formData.docPath === undefined ? '' : formData.docPath;
   var subProject = "[]";
   var subDocpath = [];
-  if (formData.subProject !== undefined) {
-    subProject = "[" + formData.subProject.join(",") + "]";
-    for(i=0; i<formData.subProject.length; i++){
-        subDocpath.push("docs");
+  if (formData.ci === true) {
+      subDocpath = "";
```
Comment (pri22296) [inline backend/generator.js]: Replace `""` with `"[]"`

Comment (sch00lb0y) [reply backend/generator.js]: updated.

**Comment by sch00lb0y** [comment]
@imujjwal96 right now.I registering ci for sub project to trigger documentation. The documentation is generated by using `yaml` config

**Comment by imujjwal96** [comment]
So that means if i dont register a repository from the UI, any changes made to that repository wouldn't trigger the CI, right?

**Comment by sch00lb0y** [comment]
@imujjwal96 Yeah

File: backend/generator.js [hunk 2]
```diff
@@ -25,10 +25,15 @@ exports.executeScript = function (socket, formData, callback) {
   var docPath = formData.docPath === undefined ? '' : formData.docPath;
   var subProject = "[]";
   var subDocpath = [];
-  if (formData.subProject !== undefined) {
-    subProject = "[" + formData.subProject.join(",") + "]";
-    for(i=0; i<formData.subProject.length; i++){
-        subDocpath.push("docs");
+  if (formData.ci === true) {
+      subDocpath = "[]";
```
Comment (imujjwal96) [inline backend/generator.js]: change this to 
subDocpath = [];

File: backend/generator.js [hunk 3]
```diff
@@ -25,10 +25,15 @@ exports.executeScript = function (socket, formData, callback) {
   var docPath = formData.docPath === undefined ? '' : formData.docPath;
   var subProject = "[]";
   var subDocpath = [];
```
Comment (pri22296) [inline backend/generator.js]: As `subDocpath` is now a string, initialize it with "[]"

Comment (sch00lb0y) [reply backend/generator.js]: @pri22296 if it is webui generator. it'll push the doc path and joins as a string `subDocpath = "[" + subDocpath.join(",") + "]";` . please check the else part

**Pull request merged**